creator-node/src/services/ContentNodeInfoManager.ts

```diff
@@ -0,0 +1,161 @@
+import type Logger from 'bunyan'
+
+import _ from 'lodash'
+
+import initAudiusLibs from './initAudiusLibs'
+import { logger as defaultLogger } from '../logging'
+import defaultRedisClient from '../redis'
+import { Redis } from 'ioredis'
+
+export type EthContracts = {
+  getServiceProviderList: (spType: string) => Promise<LibsServiceProvider[]>
+}
+export type LibsServiceProvider = {
+  owner: any // Libs typed this as any, but the contract has it as address
+  delegateOwnerWallet: any // Libs typed this as any, but the contract has it as address
+  endpoint: any // Libs typed this as any, but the contract has it as string
+  spID: number
+  type: string
+  blockNumber: number
+}
+export type ContentNodeFromChain = {
+  endpoint: string
+  delegateOwnerWallet: string
+}
+export interface IContentNodeInfoManager {
+  getMapOfSpIdToChainInfo: () => Promise<Map<number, ContentNodeFromChain>>
+  getMapOfCNodeEndpointToSpId: () => Promise<Map<string, number>>
+  getSerializedMapOfCNodeEndpointToSpId: () => Promise<string>
+  getSpIdFromEndpoint: (endpoint: string) => Promise<number | undefined>
+  getContentNodeInfoFromSpId: (
+    spId: number
+  ) => Promise<ContentNodeFromChain | undefined>
+}
+
+const SP_ID_TO_CHAIN_INFO_MAP_KEY = 'contentNodeInfoManagerSpIdMap'
+
+function ContentNodeInfoManager(
+  logger = defaultLogger,
+  redisClient = defaultRedisClient,
+  makeEthContracts: () => Promise<EthContracts> = _makeEthContracts(logger),
+  cacheTtlSeconds = 10 * 60, // 10 minutes
+  ignoreCache = false
+): IContentNodeInfoManager {
+  return {
+    async getMapOfSpIdToChainInfo(): Promise<
```

Comment:
i didn't realize you were removing the queue altogether - i'm slightly skeptical of this pattern bc it means occasionally a caller will have to wait for the chain network call, which is not ideal

i like removing the complexity of maintaining that queue, so maybe instead of expiring on TTL, we return the expired value first and then background refresh from chain? so caller never has to wait
i think there might be some redis native way to do this

Reply:
I'm not aware of a redis native way off the top of my head - I can search but when would waiting for the chain call be an issue since aren't all of the consumers of this are in background tasks?

Reply:
synced offline -- will change back to a queue